modules/caddyhttp/reverseproxy/upstreams.go

```diff
@@ -265,6 +343,34 @@ func (al aLookup) isFresh() bool {
 	return time.Since(al.freshness) < time.Duration(al.aUpstreams.Refresh)
 }
 
+// UpstreamResolver holds the set of addresses of DNS resolvers of
+// upstream addresses
+type UpstreamResolver struct {
+	// The addresses of DNS resolvers to use when looking up the addresses of proxy upstreams.
+	// It accepts [network addresses](/docs/conventions#network-addresses)
+	// with port range of only 1. If the host is an IP address, it will be dialed directly to resolve the upstream server.
+	// If the host is not an IP address, the addresses are resolved using the [name resolution convention](https://golang.org/pkg/net/#hdr-Name_Resolution) of the Go standard library.
+	// If the array contains more than 1 resolver address, one is chosen at random.
+	Addresses []string `json:"addresses,omitempty"`
+	netAddrs  []caddy.NetworkAddress
+}
+
+// ParseAddresses parses all the configured network addresses
+// and ensures they're ready to be used.
+func (u UpstreamResolver) ParseAddresses() error {
```

Comment:
This should be a pointer receiver because you're modifying the receiver and expecting the same to reflect back to the caller

Reply:
GOOD CATCH. Tricky tricky

Reply:
Nice, thanks. I keep forgetting to think about that, it's such a subtle thing in Go 🤦‍♂️